ajax/asset/assetdefinition.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+use Glpi\Asset\AssetDefinition;
+use Glpi\Asset\CustomFieldDefinition;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/** @var \Glpi\Controller\LegacyFileLoadController $this */
+$this->setAjax();
+
+Session::checkRight(AssetDefinition::$rightname, READ);
+Session::writeClose();
+
+if ($_REQUEST['action'] === 'get_all_fields') {
+    header("Content-Type: application/json; charset=UTF-8");
+    $definition = new AssetDefinition();
+    if (!$definition->getFromDB($_GET['assetdefinitions_id'])) {
+        http_response_code(404);
+        exit();
+    }
+    $all_fields = $definition->getAllFields();
+    $field_results = [];
+    foreach ($all_fields as $k => $v) {
+        $field_info = is_array($v) ? $v : ['text' => $v];
+        if (!empty($_POST['searchText']) && stripos($field_info['text'], $_POST['searchText']) === false) {
+            continue;
+        }
+        $field_info['id'] = $k;
+        $field_results[] = $field_info;
+    }
+    echo json_encode([
+        'results' => $field_results,
+        'count' => count($all_fields)
+    ], JSON_THROW_ON_ERROR);
+    exit();
+} else if ($_REQUEST['action'] === 'get_field_placeholder' && isset($_POST['fields']) && is_array($_POST['fields'])) {
+    header("Content-Type: application/json; charset=UTF-8");
+    $custom_field = new CustomFieldDefinition();
+    $results = [];
+    foreach ($_POST['fields'] as $field) {
+        if ($field['customfields_id'] > 0) {
+            if (!$custom_field->getFromDB($field['customfields_id'])) {
+                http_response_code(404);
+                exit();
```

Comment:
```suggestion
                throw new NotFoundHttpException();
```

Reply:
There is no reason to be returning a full GLPI HTML error page for AJAX requests.

Reply:
See the comment for lines 49-50.

Reply:
That wouldn't resolve my point even if it could fix this specific case. We still don't need to return an entire GLPI page for `$.load` calls or any others that would return HTML normally, but have an error and need to control the status code.

Reply:
This will be fixed by #18297.